pkg/scale/encode.go

```diff
@@ -140,7 +140,7 @@ func (es *encodeState) encodeCustomPrimitive(in interface{}) (err error) {
 	case reflect.Uint64:
 		in = reflect.ValueOf(in).Convert(reflect.TypeOf(uint64(0))).Interface()
 	default:
-		err = fmt.Errorf("unsupported type for custom primitive: %T", in)
```

Comment:
you could make a separate sentinel error for `unsupported type for custom primitive`